### PR TITLE
added support for legacy transactions

### PR DESCRIPTION
**Pull request by GandalfTheBrown**
Hey James, added legacy support so that it verifies p2sh-p2pwkh and p2sh-p2wsh!  
Also added a function to extract signature script
I tested it locally with the following transaction and it worked fine. Did not update your test file. 
      '0x0200000000010111b6e0460bb810b05744f8d38262f95fbab02b168b070598a6f31fad438fced4000000001716001427c106013c0042da165c082b3870c31fb3ab4683feffffff0200ca9a3b0000000017a914d8b6fcc85a383261df05423ddf068a8987bf0287873067a3fa0100000017a914d5df0b9ca6c0e1ba60a9ff29359d2600d9c6659d870247304402203b85cb05b43cc68df72e2e54c6cb508aa324a5de0c53f1bbfe997cbd7509774d022041e1b1823bdaddcd6581d7cde6e6a4c4dbef483e42e59e04dbacbaf537c3e3e8012103fbbdb3b3fc3abbbd983b20a557445fb041d6f21cc5977d2121971cb1ce5298978c000000'

Haven't tested the header tools yet and will update if something is breaking.

### CODE REVIEW

File: contracts/BTCUtils.sol [hunk 1]
```diff
@@ -212,7 +228,14 @@ library BTCUtils {
...
     function findNumOutputs(bytes _b) public pure returns (uint256) {
-        return 7 + (41 * extractNumInputs(_b));
+        
+        uint8 _k;
+        if (keccak256(_b.slice(43, 1)) == keccak256(hex'00')) {
+            _k = 41;
+        } else if (keccak256(_b.slice(43, 1)) == keccak256(hex'17')) {
+            _k = 64;
+        }
+        return 7 + (_k  * extractNumInputs(_b));
```
[the remark this answers is not in the document]
Comment (GandalfTheBrown) [reply contracts/BTCUtils.sol]: pushed the new code and addressed that issue. It works for mixed case now.

File: contracts/BTCUtils.sol [hunk 2]
```diff
@@ -241,8 +264,14 @@ library BTCUtils {
...
         require(_index < extractNumInputs(_b), "Index more than number of inputs");
-        uint256 _offset = 7 + (41 * _index);
-        return _b.slice(_offset, 41);
+        uint8 _k;
+        if (keccak256(_b.slice(43, 1)) == keccak256(hex'00')) {
+            _k = 41;
+        } else if (keccak256(_b.slice(43, 1)) == keccak256(hex'17')) {
+            _k = 64;
+        }
+        uint256 _offset = 7 + (_k * _index);
```
[the remark this answers is not in the document]
Comment (GandalfTheBrown) [reply contracts/BTCUtils.sol]: pushed the new code and addressed that issue. It works for mixed case now.

[the remark this answers is not in the document]
Comment (GandalfTheBrown) [reply contracts/BTCUtils.sol]: Here's an example tx

'0x01000000000106918364fc569fe8e6c44b31929f2935e862fe98c16d47067f37a0d5a7d403ed14000000001716001422db1b6ffecf633f5c4253361a4172d1c05ee0c0ffffffff1746bd867400f3494b8f44c24b83e1aa58c4f0ff25b4a61cffeffd4bc0f9ba300000000000ffffffff2c730a3be3dd0ebc27a34eacd2c66e1cc5f13c40e9463e1c081e821c907fef1f000000001716001422db1b6ffecf633f5c4253361a4172d1c05ee0c0ffffffff7e14490f6b0af140308e2cfffdfb93410ed8352cfdf599a6b0928ddee8d0a0cc000000001716001422db1b6ffecf633f5c4253361a4172d1c05ee0c0ffffffff4c619eed505a7f612ca026c62414ad482a19831bd800ed22950ae58ef838a08b000000001716001422db1b6ffecf633f5c4253361a4172d1c05ee0c0ffffffff9544893a09d109d6e836e526718ebb5946e20005ade99501bdacae15d4c45ba6000000001716001422db1b6ffecf633f5c4253361a4172d1c05ee0c0ffffffff0650c300000000000017a914e0db2f8b6e3d2503c92dd7c687d58d5355c321d68790d003000000000017a9148056042f8976f7875f5e011ff99b0a5238e72cbf87e09304000000000017a91489ebe0dd8d3e454e6bfc4c66c0863609dc5bac8c87f04902000000000017a914c18a4d4c372ba8ef6c46a1f712cd7b4f51c0146d87109802000000000017a914372977576296bbc286949da50646613b4bfa462f87d44400000000000017a9148731a867a73f604fdecf04358fca714a41e3925a870247304402207d8327dac92150ed2f95816754f4100984f56ec129f6b3dd5f370ae825b9552502205c97de59b729675f342482ba72733e01dbe6e4fef4d031933a43a55e2cdf367601210285ad9300cb60de9e1a18854c21c46c823866e1f0ebea9995de18568ac19d60a302483045022100dc0c79c8db3647d97dd56591d421b58d4e0a92b6173282aa40b67282e8b6067402207a7d152cfbafbfce3ea1dbd88a8048a4797d11e7e21364bd7bf15404bb3f4cd501210285ad9300cb60de9e1a18854c21c46c823866e1f0ebea9995de18568ac19d60a302473044022043f4cafe6b7f98eb882d3c5e1d7522b30c39f594324ddbc0588c6974c643aff3022049924028208633b52824dca9144ccbe5954cf9e761a1fd49b01062cf1bb7bec901210285ad9300cb60de9e1a18854c21c46c823866e1f0ebea9995de18568ac19d60a302483045022100d9f5ab242e50114c7673f7bd1bf7f4b2b58c192c1ff7811eab1ff8d64293be3c02200b9ac9b3aa0af509b8aa8ccafe23b57891ff3928074c4036f379c6f4cf38adbb01210285ad9300cb60de9e1a18854c21c46c823866e1f0ebea9995de18568ac19d60a302483045022100d519de61a499e136f0e4a64950a052ca2aa7d82abbc3c30da27290323e38afb602207afaea784b632e270001a22f116420b982f645d22b3c31e8403ab25f178ba5d401210285ad9300cb60de9e1a18854c21c46c823866e1f0ebea9995de18568ac19d60a300000000'

**Comment by prestwich** [comment]
@GandalfTheBrown can you add the tx above to the test scripts?

**Comment by prestwich** [comment]
Closing as stale, because v2.0.0 will be coming soon

**Pull request closed without merging**